### PR TITLE
Custom kernel for CFG, parallel execution of sampling

### PR DESCRIPTION
**Pull request by vklimkov-nvidia**
Speeding up the inference by speeding up the sampling for multi-vocab. Currently we are CPU bounded during sampling and biggest improvement comes from running single custom CFG kernel per decoder iteration instead of computing updated logits in each request separately.

### CODE REVIEW

**Comment by rmittal-github** [comment]
@styagi130 @anand-nv to review the changes
